src/ui/toolbar/GPSIndicator.qml

```diff
@@ -94,17 +94,36 @@ Item {
         anchors.left:           gpsIcon.right
 
         QGCLabel {
-            anchors.horizontalCenter:   hdopValue.horizontalCenter
-            visible:                    _activeVehicle && !isNaN(_activeVehicle.gps.hdop.value)
+            anchors.horizontalCenter:   gpsLock.horizontalCenter
+            visible:                    _activeVehicle && _activeVehicle.gps.count.valueString !== ""
             color:                      qgcPal.buttonText
             text:                       _activeVehicle ? _activeVehicle.gps.count.valueString : ""
         }
 
         QGCLabel {
-            id:         hdopValue
-            visible:    _activeVehicle && !isNaN(_activeVehicle.gps.hdop.value)
-            color:      qgcPal.buttonText
-            text:       _activeVehicle ? _activeVehicle.gps.hdop.value.toFixed(1) : ""
+            id:         gpsLock
+            visible:    _activeVehicle && _activeVehicle.gps.lock.enumStringValue !== ""
+            color:      getLockColor()
+            text:       getLockText()
+
+            function getLockColor() {
+                // Make the text orange if we don't have RTK lock
+                if (!_activeVehicle) return qgcPal.buttonText
+                const lockEnum = _activeVehicle.gps.lock.enumIndex
+                if (lockEnum === 3 || lockEnum === 4) return "orange"
+                return qgcPal.buttonText
+            }
+
+            function getLockText() {
+                // If enumIndex is 3, we have 3D lock. If enumIndex is 5 or 6, we have RTK lock.
+                // As a pilot, I don't care if it's RTK fix or RTK Float, so we just show RTK.
```

Comment:
I'm actually not sure? Should ask Tech

Reply:
Regarding if we want to display "RTK" instead of "RTK Fix" and "RTK Float"?

Reply:
I was thinking about https://aviant.atlassian.net/wiki/spaces/OP/pages/202866858/Operational+directives#Reduced-delivery-area-radius-(240116), but it seems like any RTK is sufficient. @Torjusba knows

Reply:
The work document is a bit imprecise with regards to fix/float wording. The way I read it, any RTK should be sufficient, and the pilot doesn't need to know which one it is.